Collectors/DetailedCollector.py

```diff
@@ -228,9 +237,92 @@ def addRecord(self, sid, userID, fileClose, timestamp, addr, openTime):
             self.metrics_q.put({'type': 'message sent', 'count': 1, 'message_type': 'wlcg'})
 
         return rec
-
-
-    def process_tcp(self, decoded_packet: decoding.gstream, addr: str):
+        
+    # return the VO based on the path    
+    def returnVO(self,fname):
+
+       if fname.startswith('/user'):
+            return 'user'
+       if fname.startswith('/osgconnect/public'):
+            return 'osg'
+       elif fname.startswith('/pnfs/fnal.gov/usr'):
+            return 'pnfs/fnal'
+       elif fname.startswith('/hcc'):
+            return 'hcc'
+       elif fname.startswith('/gwdata'):   
+            return 'gwdata'
+       elif fname.startswith('/chtc/'):
+            return 'chtc'
+       elif fname.startswith('/icecube/'):
+            return 'icecube'
+       else:
+            return "noVO"
+
+
+    def process_gstream(self, gstream, addr, sid):                                                      
+
+        self.logger.info("GStream")
+        hostname = ""                                                                                   
+        hostip = ""                                                                                     
+        site = ""                                                                                       
+        lcg_record = False                                                                              
+        try:                                                                                            
+            hostip = addr                                                                               
+            try:                                                                                        
+                hostname = socket.gethostbyaddr(addr)[0]                                                
+            except Exception as e:                                                                      
+                self.logger.error("No able to gethostnyname")                                           
+                                                                                                        
+            if sid in self._servers:                                                                    
+                s = self._servers[sid]
+                if s .site is not None:
+                    site = s.site.decode('utf-8')
+                                                                                                        
+                                                                      
+            for event in gstream.events:
+                try: 
+                     event["sid"] = sid
+                     event["server_ip"] = hostip
+                     event["server_hostname"] = hostname
+                     event["file_path"] = event.pop("lfn")
```

Comment:
I think it's fine, but why are you using "pop" and not just accessing the element?

Reply:
To change the "field's" name